peps/pep-0771.rst

```diff
@@ -0,0 +1,505 @@
+without any default extras. In this case, as will be shown in more detail in
+`How to teach this`_, the best approach is to define an extras which could be
+called e.g. ``minimal`` or ``nodefault`` (the naming would be up to the package
+maintainer) which would be an empty set of dependencies. If this extras is
+specified, no default extras will be included, so that e.g. ``package[minimal]``
+would include only required dependencies and no extras. Note that this requires
+no additional specification and is a natural consequence of the rule described
+in `Overriding default extras`_.
+
+There are however valid use cases where package maintainers may not want to
+provide this. For example, in the case of the multiple possible frontends or
+backends, it may be that the package would not be functional without any of the
+options. To take a specific example, a package may need either PyQt or PySide to
+be installed but will not work if none are provided, so a package maintainer may
+therefore not want to provide an option to install the package without any
+extras.
+
+Backward Compatibility
+======================
+
+All package specification cases valid under :pep:`508` will remain valid.
+Therefore, this proposal is fully backward-compatible with existing :pep:`508`
+usage.
+
+Once packages start defining default extras, those defaults will only be honored
+with recent versions of packaging tools which implement this PEP, but those
+packages will remain fully backward-compatible with older packaging tools - with
+the only difference that the default extras will not be installed automatically
+when older packaging tools are used.
+
+The only conceptual backward-compatibility issue to consider is the fact that
+this PEP changes extras to no longer be strictly additive, in that specifying
+an extra such as ``minimal`` could result in fewer packages being installed.
+
+Security Implications
+=====================
+
+There are no known security implications for this PEP.
+
+How to teach this
+=================
+
+The rule above regarding only installing default extras when no extras
+are explicitly specified, combined with the introduction of the
+``Default-Extra:`` keyword and ``default-optional-dependencies`` metadata key
+allows us to address several different use cases. Below we take a look at the
+two specific use cases raised in the `Motivation`_ section and how package
+maintainers should be taught to address these.
+
+Recommended dependencies and minimal installations
+--------------------------------------------------
+
+First, we consider the case of packages that want recommended
+but not strictly required dependencies installed by default, while also
+providing a way to only install the required dependencies.
+
+In order to do this, a package maintainer would define an extras called
+``recommended`` containing the recommended but not required dependencies, and
+would choose to have this be included as a default extras:
```

Comment:
```suggestion
In order to do this, a package maintainer would define extras called
``recommended`` containing the recommended but not required dependencies, and
would choose to have this be included as default extras:
```

Reply:
I have fixed this by changing an extras -> an extra